### PR TITLE
chapter_9

### PR DESCRIPTION
**Pull request by muizzuddinyafiq**
ready

### CODE REVIEW

**Comment by muizzuddinyafiq** [comment]
ready

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,18 +1,20 @@
 class SessionsController < ApplicationController
-  def new; end
+  def new
+  end
```
Comment (ghost) [inline app/controllers/sessions_controller.rb]:
```suggestion
  def new; end
```

**Review by ghost — COMMENTED** [review]
cmt

**Comment by muizzuddinyafiq** [comment]
ready

File: app/models/user.rb [hunk 2]
```diff
@@ -1,12 +1,37 @@
...
   has_secure_password
   validates :password, presence: true, length: { minimum: 6 }
+
+  class << self
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+
+  # Returns the hash digest of the given string.
+  def digest string
```
Comment (huongnt-2545) [inline app/models/user.rb]: indent from line 18 -> 26

**Comment by muizzuddinyafiq** [comment]
ready

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -4,15 +4,42 @@ def log_in user
...
     @current_user = nil
   end
+
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def authenticated? remember_token
```
Comment (ngocvt-0484) [inline app/helpers/sessions_helper.rb]: ? 
this method shouldn't define here
please check again

Comment (muizzuddinyafiq) [reply app/helpers/sessions_helper.rb]: Screenshot from 2021-09-10 16-51-24

On the video it is define in the session_helper, I saw others people also same like me

**Comment by muizzuddinyafiq** [comment]
ready

File: app/helpers/sessions_helper.rb (hunk 3 above)
Comment (ngocvt-0484) [reply app/helpers/sessions_helper.rb]: delete this function please
because it defined on user.rb model

**Comment by muizzuddinyafiq** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**